smt.go

```diff
@@ -123,6 +123,20 @@ func (smt *SparseMerkleTree) GetForRoot(key []byte, root []byte) ([]byte, error)
 	return value, nil
 }
 
+// Has returns true if tree cointains given key, false otherwise.
+func (smt *SparseMerkleTree) Has(key []byte) (bool, error) {
+	val, err := smt.Get(key)
+	// this covers both val == nil and val == defaultValue
+	return len(val) > 0, err
```

Comment:
I think this should do a byte comparison with defaultValue instead of a length check, as if defaultValue is changed this would break?

Reply:
Yes, I guess it's a premature optimization in my code. I don't think of any reason to change `defaultValue` to something other, but semantically `Has` should compare with `defaultValue`.